test/filters-cases/sass-square.asm.binary.directives.labels.comments.approved.txt

```diff
@@ -9,14 +9,7 @@
       "address": 8,
       "labels": [],
       "opcodes": [
-        "e4",
```

Comment:
can you make sure this is kept the way it is?

Reply:
Is there a particular reason why SASS opcodes need to be split per-byte?
GPU instructions are 32 or 64-bit. That's what nvdisasm shows and that's how other disassemblers in the compiler explorer behave. E.g. https://godbolt.org/z/hhxrEecqM

Reply:
All opcodes are split byte per byte, because that's how CPU's work and most compilers currently compile to CPU opcodes.

If you want to change the GPU approach, I would very much appreciate a GH issue first before actually changing the way we show it to the user. (we haven't heard any complaints afaik so far)

Reply:
I've opened an issue as you've requested: https://github.com/compiler-explorer/compiler-explorer/issues/2760

IMO it would make sense not to assume particular opcode format and trust the disassembler to provide a sensible representation. This patch actually does exactly that for SASS and AMDGPU disassembly. IMO removing the current code which special-cases nvdisasm is a clean up.

As for no complaints, there are probably not that many users who care about SASS disassembly in general and few of those would actually care about SASS opcode encoding in particular. I personally don't and I do use compiler explorer on CUDA all the time.

Reply:
Hi both! I agree with @partouf's instincts that any change to these things should be carefully considered. That said I think @Artem-B 's argument for them being "natural-sized" is compelling. Certainly I'd expect ARM (which you linked) to be output in words and halfwords (as that's the fixed instruction size).

I think on balance we _should_ go with the change Artem-B proposes here. Will comment in the issue.